addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1028,8 +1039,23 @@ options.registry.layout_column = options.Class.extend({
      * @see this.selectClass for parameters
      */
     selectCount: async function (previewMode, widgetValue, params) {
+        const previousNbColumns = this.$('> .row').children().length;
+        let $row = this.$('> .row');
+        if (!$row.length) {
+            $row = this.$target.contents().wrapAll($('<div class="row"><div class="col-lg-12"/></div>')).parent().parent();
+        }
+
         const nbColumns = parseInt(widgetValue);
-        await this._updateColumnCount(nbColumns - this.$target.children().length);
+        await this._updateColumnCount($row, (nbColumns || 1) - $row.children().length);
+        // Yield UI thread to wait for event to bubble before activate_snippet is called.
+        // In this case this lets the select handle the click event before we switch snippet.
+        await new Promise(resolve => setTimeout(resolve));
```

Comment:
@sdegueldre Seems legit following your explanation. Maybe this could be done more generically though? In the activate_snippet handler at least, before option method call maybe?

Reply:
Alright, pushed the following changes:

* Using the above code in `_onActivateSnippet` handler with a more detailed comment
* Changed `_onDeactivateSnippet` to call `_onActivateSnippet` instead of calling `_activateSnippet` directly
* Changed the `removeSnippet` method to take an additional parameter. Historically, it would only be called by the left panel trash can, and so it would always need to activate the parent of the removed snippet. Since the column option now also fires a `remove_snippet` event to remove some of its contents, we don't always want to activate the parent snippet, so you can now call it with a parameter to tell it to keep the current active snippet. This was causing other issues.

also left a few TODO comments for things to check with you.

Reply:
@qsm-odoo actually this needs some more work. I've ran into a few issues that are still not sorted out. In `removeSnippet`, the editor calls its destroy method, but the snippetsMenu keeps a reference to it in its `snippetEditors` field, and keeps trying to toggle its options after its been destroyed. At the moment, this is silent but as soon as snippetEditors start removing their `$optionsSection` properly, this will crash because `_activateSnippet` will try to call `toggleOptions(false)` on all editors (even destroyed ones), which tries to get all editors on its `customize$Elements` through jQuery's `data` method, but because the editor destroyed its `$optionsSection` (which is a `customize$Element`) it will throw. Asking the `SnippetsMenu` to destroy the editor instead (allowing it to remove it from its `snippetEditors` field) prevents the crash, but I still need to solve the "also delete all parents that are now empty" problem.

Reply:
@qsm-odoo as discussed in person, since making things more generic is more involved than we previously thought, I'll just go with the initial version of using setTimeout directly in the option method, and we'll improve the underlying system separately. If there is nothing else this PR is ready :+1: